server/websocket.go

```diff
@@ -987,6 +988,39 @@ func (p *Plugin) WebSocketMessageHasBeenPosted(connID, userID string, req *model
 			return
 		}
 		msg.Data = []byte(msgData)
+	case clientMessageTypeCaption:
+		// Sent from the transcriber.
+		sessionID, ok := req.Data["session_id"].(string)
+		if !ok {
+			p.LogError("invalid or missing session_id in caption ws message")
+			return
+		}
+		userID, ok := req.Data["user_id"].(string)
+		if !ok {
+			p.LogError("invalid or missing user_id in caption ws message")
+			return
+		}
+		text, ok := req.Data["text"].(string)
+		if !ok {
+			p.LogError("invalid or missing text in caption ws message")
+			return
+		}
+		newAudioLenMs, ok := req.Data["new_audio_len_ms"].(float64)
+		if !ok {
+			p.LogError("invalid or missing new_audio_len_ms in caption ws message")
+			return
+		}
+		p.handleCaptionMessage(us.channelID, sessionID, userID, text, newAudioLenMs)
+		return
+	case clientMessageTypeMetric:
+		// Sent from the transcriber.
```

Comment:
I think this is great and could be sent by any client eventually to track more stuff (e.g. WebRTC call quality as seen from the client side or more client side performance data).